PanCardView.iOS/CardsViewRenderer.cs

```diff
@@ -119,31 +120,42 @@ private void OnSwiped(UISwipeGestureRecognizer gesture)
                         : ItemSwipeDirection.Down;
 
             Element?.OnSwiped(swipeDirection);
-        }
-
-        private bool ShouldBeRequiredToFailBy(UIGestureRecognizer gestureRecognizer, UIGestureRecognizer otherGestureRecognizer)
-            => IsPanGestureHandled() && otherGestureRecognizer.View != this;
-
-        private bool ShouldRecognizeSimultaneously(UIGestureRecognizer gestureRecognizer, UIGestureRecognizer otherGestureRecognizer)
-        {
-            if (!(gestureRecognizer is UIPanGestureRecognizer panGesture))
-            {
-                return true;
-            }
-
-            var parent = Element?.Parent;
-            while (parent != null)
-            {
-                if (parent is FlyoutPage && (Element?.IsHorizontalOrientation ?? false))
-                {
-                    var velocity = panGesture.VelocityInView(this);
-                    return Abs(velocity.Y) > Abs(velocity.X);
-                }
-                parent = parent.Parent;
-            }
-            return !IsPanGestureHandled();
-        }
-
+        }
+
+        private bool ShouldBeginPan(UIGestureRecognizer recognizer)
+        {
+            if (recognizer is UIPanGestureRecognizer pangesture && !Element.IsVerticalSwipeEnabled)

```

Comment:
?

```csharp
if (recognizer is UIPanGestureRecognizer pangesture) {
    var velocity = pangesture.VelocityInView(this);
    var absVelocityX = Abs(velocity.X);
    var absVelocityY = Abs(velocity.Y);
    return (absVelocityY < absVelocityX && Element.IsHorizontalOrientation) ||
               (absVelocityY > absVelocityX && !Element.IsHorizontalOrientation)
}

return true;
```

Reply:
But the problem is that we don't want to fail panGesture in case of vertical move if it's not needed (There is not ScrollView / CollectionView / ListView etc.)

Maybe this code should be done it ShouldBeRequiredToFailBy?

Also, could you please rename ShouldBeginPan to ShouldBegin

Reply:
So wouldn't adding the original !Element.IsVerticalSwipeEnabled check to your suggested code cater for your concern about not failing a vertical pan unnecessarily?

Reply:
The suggested code will only fail a vertical pan if it's mostly horizontal though right? Is that a problem?